src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -2,15 +2,19 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-
+import { IAction } from 'vs/base/common/actions';
 import { timeout } from 'vs/base/common/async';
 import { debounce } from 'vs/base/common/decorators';
 import { Emitter } from 'vs/base/common/event';
+import { localize } from 'vs/nls';
 import { ILogService } from 'vs/platform/log/common/log';
 import { ICommandDetectionCapability, TerminalCapability, ITerminalCommand, IHandleCommandOptions, ICommandInvalidationRequest, CommandInvalidationReason, ISerializedCommand, ISerializedCommandDetectionCapability } from 'vs/platform/terminal/common/capabilities/capabilities';
+//TODO
```

Comment:
perhaps this should be deferred to the terminal instance as well

Reply:
They should get registered into the terminal instance, we may want the main list that applies to all instances to live in terminal service or terminal instance service so they can be initialized when the instance is created?

Reply:
Alternatively require any consumers to register them into each newly created terminal instance.

Reply:
wouldn't it make more sense to keep the registration on the `contextActionAddon` which is accessible via the terminal instance by way of `xterm.contextAction`?